ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx

```diff
@@ -16,14 +16,17 @@ import PencilIcon from 'mdi-material-ui/PencilOutline';
 import AddPanelGroupIcon from 'mdi-material-ui/PlusBoxOutline';
 import AddPanelIcon from 'mdi-material-ui/ChartBoxPlusOutline';
 import { ErrorBoundary, ErrorAlert } from '@perses-dev/components';
-import { useDashboardActions, useEditMode } from '../../context';
+import { UseMutationResult } from '@tanstack/react-query';
+import { DashboardResource } from '@perses-dev/core';
+import { useDashboard, useDashboardActions, useEditMode } from '../../context';
 import { TemplateVariableList } from '../Variables';
 import { TimeRangeControls } from '../TimeRangeControls';
 import { DownloadButton } from '../DownloadButton';
 
 export interface DashboardToolbarProps {
   dashboardName: string;
   dashboardTitleComponent?: JSX.Element;
+  dashboardMutation?: UseMutationResult<DashboardResource, Error, DashboardResource>;
```

Comment:
Instead of adding a dashboardMutation prop here, I think it's better if we add an onSave prop that returns a promise so this component doesn't need to know anything about react-query and we can also handle loading/error/success state in here as well.

For example:

```
const { dashboard } = useDashboard();
const onSaveButtonClick = async () => {
    if (props.onSave) {
      try {
        await props.onSave(dashboard);
      } catch (err) {
        exceptionSnackbar(
          err,
          'There was an error while trying to save the dashboard. Please try again.'
        );
      }
    }
  };
```

Reply:
+1 `onSaveClick` to keep it generic. @eunicorn As far as the promise goes, are you thinking that when the Promise is in pending state, the button would be disabled and show a loading icon?

Reply:
while I agree with this statement (dashboard shouldn't depend on a third party library like react-query), I just would like to raise the point `@perses-dev/dashboard` has a direct dependency to `@tanstack/react-query`.

So from my point of view: 
1. Yes it makes sense to avoid having a property that is using the type from `@tanstack/react-query`
2. Point 1 really makes sense if the package `@perses-dev/dashboard` doesn't depend at all of `@tanstack/react-query`

Reply:
I suppose it's less about dependencies but more about interfaces. A package can transparently depend on another package but its interface is still decoupled (not saying we are prefect at this everywhere). I think this is an area where it can be decoupled.

Reply:
> exceptionSnackbar(
          err,
          'There was an error while trying to save the dashboard. Please try again.'
        );

Currently the snackbar is only available in the app.
I'm not sure if it makes sense to move it to the package `@perses-dev/component`. That would makes my life easier for sure, but then that would mean to be able to used the dashboard you need to use the `SnackbarProvider`. Maybe we don't want that ?

And if we don't want that, do you have an idea on how I could manage the error or the success message ?
I really like the snackbar because it's perfect to display a volatile message

Reply:
It would look something like this:

```tsx
<DashboardToolbar onSave={async (dashboardJSON) => {
    showLoadingSnackbar();
    try {
        const result = await saveDashboard() // via backend
        showSuccessSnackbar();
    } catch(e) {
        showErrorSnackbar(e);
    }
}} />
```